src/CommonITILTask.php

```diff
@@ -291,10 +291,55 @@ private function handleTaskDuration(array &$input, int $timestart, int $timeend)
         }
     }
 
+    public function assignTechFromtask($input)
+    {
+        //if user or group assigned to ticket task, add it to the ticket
+        $entities = [
+            'Ticket' => [
+                'users_id_tech' => new Ticket_User(),
+                'groups_id_tech' => new Group_Ticket()
+            ],
+            'Change' => [
+                'users_id_tech' => new Change_User(),
+                'groups_id_tech' => new Change_Group()
+            ],
+            'Problem' => [
+                'users_id_tech' => new Problem_User(),
+                'groups_id_tech' => new Group_Problem()
+            ],
+        ];
+        foreach ($entities as $entityType => $entityData) {
+            $entityType = strtolower($entityType);
```

Comment:
I suppose you are iterating on every possible parent type here because there is no method to get the final ITIL class for a given task class.

I think it would be clearer to add a abstract `getTargetITILClass` method to the `CommonITILTask` class.
Each child class would then define its correct target (e.g. `return Ticket::class` for `TicketTask`).

With this new method you don't need to loop on types anymore and can just call `$this->getTargetITILClass()`.